flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/BoundedBlockingSubpartitionReader.java

```diff
@@ -89,9 +114,30 @@ public BufferAndBacklog getNextBuffer() throws IOException {
 		return BufferAndBacklog.fromBufferAndLookahead(current, nextBuffer, dataBufferBacklog);
 	}
 
+	/**
+	 * This method is actually only meaningful for the {@link BoundedBlockingSubpartitionType#FILE}. For the
+	 * other types the next buffer is always not null while {@link ResultSubpartitionView#getNextBuffer()},
+	 * so it is no need to notify available via this method. But the implementation is also compatible with
+	 * other types even though called by mistake.
+	 */
 	@Override
 	public void notifyDataAvailable() {
-		throw new IllegalStateException("No data should become available on a blocking partition during consumption.");
+		if (nextBuffer == null) {
```

Comment:
add `checkState` for `BoundedBlockingSubpartitionType#FILE`?

Reply:
Do you mean checking via `checkState(dataReader instanceof FileChannelBoundedData.FileBufferReader)` when next buffer is null?